core/slim_sim.cpp

```diff
@@ -3488,6 +3506,76 @@ void SLiMSim::_CheckMutationStackPolicy(void)
 
 // other tree sequence methods should probably be implemented here, unless you just make them inline forwards to code in treerec...
 
+void
+SLiMSim::handle_error(std::string msg, int err)
+{
+	std::cout << "Error:" << msg << ":" << msp_strerror(err) << std::endl;
+	EIDOS_TERMINATION << msg << EidosTerminate();
+	exit(1);
+}
+
+
+void SLiMSim::simplifyTables(void){
+
+	std::map<slim_objectid_t,Subpopulation*>::iterator it;
+	std::vector<Individual*> subpopulationIndividuals;
+	std::vector<Individual*> populationIndividuals;
+	std::vector<node_id_t> samples;
+	std::map<int,node_id_t> newMap;
+	
+	for ( it = population_.begin(); it != population_.end(); it++){
+		subpopulationIndividuals = it->second->parent_individuals_;
+		populationIndividuals.insert(populationIndividuals.end(), subpopulationIndividuals.begin(), subpopulationIndividuals.end());
+	}
+	
+	slim_pedigreeid_t IndID;
+	int G1;
+	int G2;
+	int newValueInNodeTable = 0;
+	for (unsigned i; i < populationIndividuals.size(); i++){
+		IndID = populationIndividuals[i]->PedigreeID();
+		G1 = 2 * IndID;
+		G2 = G1 + 1;
+
+		samples.push_back(getMSPID(G1));
+		samples.push_back(getMSPID(G2));	
+				
+		newMap[G1] = (node_id_t)newValueInNodeTable++;
+		newMap[G2] = (node_id_t)newValueInNodeTable++;
+	}
+
+	ret = sort_tables(&nodes, &edges, &migrations, &sites, &mutations, 0);				
+	if (ret < 0) {
+		handle_error("sort_tables", ret);
+	}
+	
+	ret = simplifier_alloc(&simplifier,(double)chromosome_.last_position_ + 1, samples.data(), samples.size(),
+	    &nodes, &edges, &migrations, &sites, &mutations, 0, 0);
+	if (ret < 0) {
+		handle_error("simplifier_alloc", ret);
+	}
+
+	ret = simplifier_run(&simplifier, NULL);
+	if (ret < 0) {
+		handle_error("simplifier_run", ret);
+	}
+
+	simplifier_free(&simplifier);
+	
+	FSIDAS = (int)((CurrentTreeSequenceIndividual->PedigreeID()) * 2) + 2;
```

Comment:
the +2 implies that it will be the next one generated. `(CurrentTreeSequenceIndividual->PedigreeID() * 2) +1` should have been the last genome generated before the `SimplifyTables()` gets called.

Reply:
ah, gotcha.